src/index.ts

```diff
@@ -129,26 +135,20 @@ export const Permit = ({ loggedInUser, userAttributes = {}, backendUrl, defaultA
   };
 
   const getCaslJson = () => {
-    // console.debug(permitCaslState);
     return permitCaslState;
   };
 
-  const check = (action: string, resource: string, resourceAttributes: Record<string, any> = {}): boolean => {
+  const check = (action: string, resource: string | ReBACResourceSchema, resourceAttributes: Record<string, any> = {}): boolean => {
     const key = generateStateKey(action, resource, resourceAttributes);
     return permitLocalState[key] ?? defaultAnswerIfNotExist;
   };
 
-  const addKeyToState = async (action: string, resource: string, resourceAttributes: Record<string, any> = {}) => {
-    const permission = await getPermissionFromBE(backendUrl, loggedInUser, action, resource, defaultAnswerIfNotExist);
+  const addKeyToState = async (action: string, resource: string | ReBACResourceSchema, resourceAttributes: Record<string, any> = {}) => {
+    const resourceKey = typeof resource === 'string' ? resource : `${resource.type}:${resource.key}`;
+    const permission = await getPermissionFromBE(backendUrl, loggedInUser, action, resourceKey, defaultAnswerIfNotExist);
     await updatePermissionState({ action, resource, userAttributes: permitState.userAttributes, resourceAttributes }, permission);
   };
 
-  const reset = () => {
```

Comment:
Why have you reverted the reset func?
Probably merge small issue
Please re-add it